File: 2568-minimum-impossible-or/2568-minimum-impossible-or.cpp
class Solution {
public:
    int minImpossibleOR(vector<int>& nums) {
        
        /*
        sort(nums.begin(),nums.end());
        
        int ans = 1;
        
        for(auto i : nums){
            if(ans == i)
                ans *= 2;
        }
        
        return ans;
        */
         unordered_set<int> s(nums.begin(), nums.end());
        int a = 1;
        while (s.count(a))
            a <<= 1;
        return a;
    }
};

File: 2568-minimum-impossible-or/2568-minimum-impossible-or.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minImpossibleOR = function(nums) {
    
    nums.sort((a,b)=> a - b);
    let ans = 1;
    nums.map((e)=>{
        if(ans === e) ans *= 2;
    })
    
    return ans;
};